src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,115 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Loader } from '../Loader';
 import { NewCommentForm } from '../NewCommentForm';
+import { getPostDetails } from '../../api/posts';
+import { getPostComments, deleteComment, addCommentToServer } from '../../api/comments';
 import './PostDetails.scss';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+type Props = {
+  selectedPostId: number;
+};
+
+export const PostDetails: React.FC<Props> = (props) => {
+  const { selectedPostId } = props;
+  const [details, setDetails] = useState({} as Post);
+  const [comments, setComments] = useState([] as Comment[]);
+  const [commentsDisplay, setCommentsDisplay] = useState(true);
+  const [loading, setLoadingStatus] = useState(true);
+
+  useEffect(() => {
+    getPostDetails(selectedPostId)
+      .then(response => {
+        setDetails(response);
+        setLoadingStatus(false);
+      });
+  }, [selectedPostId]);
+
+  useEffect(() => {
+    getPostComments(selectedPostId)
+      .then(response => setComments([...response]));
+  }, [selectedPostId]);
```

Comment:
why do you use 2 useEffects with the same dependency array?
and try to use async/await